core/src/main/java/org/commonjava/indy/core/content/DefaultDirectContentAccess.java

```diff
@@ -40,21 +47,60 @@
     @Inject
     private DownloadManager downloadManager;
 
+
     public DefaultDirectContentAccess(){}
 
-    public DefaultDirectContentAccess( DownloadManager downloadManager )
+    public DefaultDirectContentAccess( final DownloadManager downloadManager )
     {
         this.downloadManager = downloadManager;
     }
 
+
+    @Override
     public List<Transfer> retrieveAllRaw( final List<? extends ArtifactStore> stores, final String path,
                                           final EventMetadata eventMetadata )
             throws IndyWorkflowException
     {
-        final List<Transfer> txfrs = new ArrayList<Transfer>();
+        Logger logger = LoggerFactory.getLogger( getClass() );
+        ExecutorService executor = Executors.newCachedThreadPool();
```

Comment:
Using @WeftManaged @ExecutorConfig(..) allows us to tune this thread pool in Indy config files, fwiw. Also, would it make sense to use [ExecutorCompletionService](http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ExecutorCompletionService.html) here?